### PR TITLE
[ASCII-1843] Ensure configcheck result is written to stdout

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

On the work done on this PR https://github.com/DataDog/datadog-agent/pull/25595 we modified the configcheck command to ensure scrubbing would happen in the server an not in the client. 

On that PR we missed a tiny detail for the cluster agent. We forgot to write the result from communicating with the server 🤦 

In this PR we ensure that we write the result from the server into the writer, and that the CLI flag for color `-n` is respected

### Describe how to test/QA your changes

- Ensure the main agent flare archive includes the `config-check.log` file with the right information. Ensure the output does not have ANSI color strings
- Ensure the cluster agent flare archive includes the `config-check.log` file with the right information. Ensure the output does not have ANSI color strings
- Ensure CLI command `cluster-agent configcheck` works as expected. The output must include colors 
- Ensure CLI command `cluster-agent configcheck -n` works as expected. The output must not include colors 
- Ensure CLI command `agent configcheck` works as expected. The output must include colors 
- Ensure CLI command `agent configcheck -n` works as expected. The output must not include colors

### CODE REVIEW

**Review by dustmop — APPROVED** [review]
LGTM

File: pkg/flare/config_check.go [hunk 1]
```diff
@@ -50,5 +50,7 @@ func GetClusterAgentConfigCheck(w io.Writer, withDebug bool) error {
 		return fmt.Errorf("failed to query the agent (running?): %s", err)
 	}
 
-	return nil
+	_, err = w.Write(r)
```
[the remark this answers is not in the document]
Comment (GustavoCaso) [reply pkg/flare/config_check.go:53]: I added some naive e2e test for the configcheck for cluster agent https://github.com/DataDog/datadog-agent/pull/26435/commits/a75c5334c7ca4f2d04fcc8c2477a36bf2ce24086

File: test/new-e2e/tests/containers/k8s_test.go [hunk 2]
```diff
@@ -247,6 +248,27 @@ func (suite *k8sSuite) TestVersion() {
...
+		})
+		if suite.NoError(err) && len(linuxPods.Items) >= 1 {
+			stdout, stderr, err := suite.podExec("datadog", linuxPods.Items[0].Name, "cluster-agent", []string{"agent", "configcheck"})
+			if suite.NoError(err) {
+				suite.Empty(stderr, "Standard error of `agent configcheck` should be empty")
+				suite.Contains(stdout, "=== kubernetes_apiserver check ===")
+			} else {
+				suite.T().Errorf("No cluster agent pod found. %v\n", err)
+			}
+		}
```
Comment (L3n41c) [inline test/new-e2e/tests/containers/k8s_test.go:268]: We can leverage the fact that the `require` package makes the test fail early and return as soon as an error is encountered to get rid of the nested `if`.

```suggestion
		suite.Require().NoError(err)
		suite.Require().Len(linuxPods.Items, 1)
		stdout, stderr, err := suite.podExec("datadog", linuxPods.Items[0].Name, "cluster-agent", []string{"agent", "configcheck"})
		suite.Require().NoError(err)
		suite.Empty(stderr, "Standard error of `agent configcheck` should be empty")
		suite.Contains(stdout, "=== kubernetes_apiserver check ===")
```

File: pkg/cli/subcommands/dcaconfigcheck/command.go [hunk 3]
```diff
@@ -60,5 +63,14 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 }
 
 func run(_ log.Component, _ config.Component, cliParams *cliParams) error {
-	return flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose)
+	var b bytes.Buffer
+	color.Output = &b
+
+	err := flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose)
+	if err != nil {
+		return fmt.Errorf("the agent ran into an error while checking config: %v", err)
```
Comment (L3n41c) [inline pkg/cli/subcommands/dcaconfigcheck/command.go:71]:
```suggestion
		return fmt.Errorf("the agent ran into an error while checking config: %w", err)
```

File: pkg/cli/subcommands/dcaconfigcheck/command.go [hunk 4]
```diff
@@ -60,5 +63,14 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 }
 
 func run(_ log.Component, _ config.Component, cliParams *cliParams) error {
-	return flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose)
+	var b bytes.Buffer
+	color.Output = &b
+
+	err := flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose)
+	if err != nil {
```
Comment (L3n41c) [inline pkg/cli/subcommands/dcaconfigcheck/command.go:70]:
```suggestion
	if err := flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose); err != nil {
```

**Comment by GustavoCaso** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=on --variable RUN_UNIT_TESTS=on --variable RUN_KMT_TESTS=on

File: pkg/cli/subcommands/dcaconfigcheck/command.go [hunk 5]
```diff
@@ -60,5 +63,13 @@ func MakeCommand(globalParamsGetter func() GlobalParams) *cobra.Command {
 }
 
 func run(_ log.Component, _ config.Component, cliParams *cliParams) error {
-	return flare.GetClusterAgentConfigCheck(color.Output, cliParams.verbose)
+	var b bytes.Buffer
+	color.Output = &b
```
Comment (davidor) [inline pkg/cli/subcommands/dcaconfigcheck/command.go:67]: Not sure if the color part is correct. The output shows some colors in 7.54.0, but with a custom image built from this branch there are no colors.

Comment (GustavoCaso) [reply pkg/cli/subcommands/dcaconfigcheck/command.go]: @davidor I fixed on this commit https://github.com/DataDog/datadog-agent/pull/26435/commits/58a34dc5a17e93ba1d68ba895cfa8007d22f5329. Let me know what you think 😄

**Review by paulcacheux — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Review by daniel-taf — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]

**Review by davidor — APPROVED** [review]

**Review by BaptisteFoy — APPROVED** [review]

**Pull request merged**